test/chaat/handler/handler_test.clj

```diff
@@ -58,18 +82,27 @@
             response (handler/delete-user datasource request)]
         (is (= 400 (:status response)))))
 
-    (testing "Username parameter present in request,
-              but user does not exist, delete fails with 404 not found"
-      (let [params {:username "john"}
-            request {:params params}
-            response (handler/delete-user datasource request)]
-        (is (= 404 (:status response)))))
+    (let [username "john"
+          password "12345678"
+          params {:username username}
+          identity {:username username}]
 
-    (testing "Username param in request, and user exists, delete succeeds"
-      (let [username "john"
-            password "12345678"
-            _ (handler/signup datasource {:params {:username username :password password}})
-            params {:username "john"}
-            request {:params params}
-            response (handler/delete-user datasource request)]
-        (is (= 200 (:status response)))))))
+      ;; create user john
+      (let [_ (handler/signup datasource {:params {:username username :password password}})]
+        (testing "Unauthorized request, delete fails"
+          (let [request {:params params}
+                response (handler/delete-user datasource request)]
+            (is (= 401 (:status response)))))
+
+        (testing "Authorized request, delete succeeds"
+          (let [request {:params params
+                         :identity identity}
+                response (handler/delete-user datasource request)]
+            (is (= 200 (:status response)))))
+
+        (testing "Authorized request, but user does not exist"
+          (let [params {:username "john"}
+                request {:params params
+                         :identity identity}
+                response (handler/delete-user datasource request)]
+            (is (= 404 (:status response)))))))))
```

Comment:
Could use another test to verify that expired tokens (7+ days old) fail authorization checks

Reply:
This would be a good test to add. I am parking chaat now, so will not make this change, but will keep this in mind for future projects.